nemo_text_processing/inverse_text_normalization/es/taggers/electronic.py

```diff
@@ -51,25 +75,60 @@ def __init__(self):
             + pynutil.insert("\"")
         )
         single_alphanum = pynini.closure(alpha_num + delete_extra_space) + alpha_num
-        server = single_alphanum | pynini.string_file(get_abs_path("data/electronic/server_name.tsv")).invert()
-        domain = single_alphanum | pynini.string_file(get_abs_path("data/electronic/domain.tsv")).invert()
+
+        server_names = pynini.string_file(get_abs_path("data/electronic/server_name.tsv")).invert()
+        if input_case == INPUT_CASED:
+            server_names = capitalized_input_graph(server_names)
+        server = single_alphanum | server_names | pynini.closure(NEMO_ALPHA, 2)
+
+        if input_case == INPUT_CASED:
+            domain = []
+            # get domain formats
+            for d in load_labels(get_abs_path("data/electronic/domain.tsv")):
+                domain.extend(get_various_formats(d[0]))
+            domain = pynini.string_map(domain).optimize()
+        else:
+            domain = pynini.string_file(get_abs_path("data/electronic/domain.tsv")).invert()
+
+        domain = pynutil.add_weight(single_alphanum, weight=-0.0001) | domain | pynini.closure(NEMO_ALPHA, 2)
+
         domain_graph = (
             pynutil.insert("domain: \"")
             + server
-            + delete_extra_space
-            + process_dot
-            + delete_extra_space
+            + ((delete_extra_space + process_dot + delete_extra_space) | alternative_dot)
             + domain
             + pynutil.insert("\"")
         )
-        graph = (
-            username + delete_extra_space + pynutil.delete("arroba") + insert_space + delete_extra_space + domain_graph
-        )
+
+        at = pynini.accep("arroba")
+        if input_case == INPUT_CASED:
+            at |= pynini.accep("Arroba")
+
+        graph = username + delete_extra_space + pynutil.delete(at) + insert_space + delete_extra_space + domain_graph
 
         ############# url ###
-        protocol_end = pynini.cross(pynini.union("www", "w w w", "doble ve doble ve doble ve"), "www")
-        protocol_start = pynini.cross(pynini.union("http", "h t t p", "hache te te pe"), "http")
-        protocol_start |= pynini.cross(pynini.union("https", "h t t p s", "hache te te pe ese"), "https")
+        if input_case == INPUT_CASED:
```

Comment:
Question: do native speakers actually use Spanish letter names or is there a tendency to switch to English letters in this scenario?